blocklylib-core/src/main/java/com/google/blockly/android/BlockListUi.java

```diff
@@ -0,0 +1,49 @@
+package com.google.blockly.android;
+
+import android.support.annotation.Nullable;
+
+import com.google.blockly.android.control.BlocklyController;
+import com.google.blockly.android.ui.FlyoutCallback;
+import com.google.blockly.model.FlyoutCategory;
+
+/**
+ * An interface that specifies the actions that can be taken on a flyout ui component.
+ */
+public interface BlockListUI {
+
+    /**
+     * @return True if this ui is currently visible, false otherwise.
+     */
+    boolean isOpen();
+
+    /**
+     * Sets the Flyout's current {@link FlyoutCategory}, including opening or closing the drawer.
+     * In closeable toolboxes, {@code null} {@code category} is equivalent to closing the drawer.
+     * Otherwise, the drawer will be rendered empty.
+     *
+     * @param category The {@link FlyoutCategory} with blocks to display.
+     */
+    void setCurrentCategory(@Nullable FlyoutCategory category);
```

Comment:
FOR FUTURE: Since we have close() / closeBlocksDrawer(), category could be @Nonnull.